--- check_apcups.orig	2013-01-16 16:35:04.000000000 +0000
+++ check_apcups	2013-01-16 16:36:16.000000000 +0000
@@ -1,5 +1,7 @@
 #!/bin/sh
 #
+# perfData patch 0.1 MarcoMC@Inviqa.com 20130116
+# 
 # check_apcupsd 2.5
 # Nagios plugin to monitor APC Smart-UPSes and Back-UPSes using apcupsd.
 #
@@ -244,7 +246,7 @@
 		STATUS="OK"
 		EXITVALUE=0
 	fi
-	echo "${STATUS} - Battery Charge: ${VALUE}%|'Battery Charge'=${VALUE}%;$WARNVAL:;$CRITVAL:;;"
+	echo "${STATUS} - Battery Charge: ${VALUE}%|'Battery Charge'=${VALUE}%;$WARNVAL;$CRITVAL;0;100"
 	;;
 itemp)
 	if [ "$CRITVAL" = "" ]; then
@@ -263,7 +265,7 @@
 		STATUS="OK"
 		EXITVALUE=0
 	fi
-	echo "${STATUS} - Internal Temperature: $VALUE C|'Battery Temperature'=${VALUE}C;$WARNVAL;$CRITVAL;;"
+	echo "${STATUS} - Internal Temperature: $VALUE C|'Battery Temperature'=${VALUE}C;$WARNVAL;$CRITVAL;0;100"
 	;;
 loadpct)
 	if [ "$CRITVAL" = "" ]; then
@@ -282,7 +284,7 @@
 		STATUS="OK"
 		EXITVALUE=0
 	fi
-	echo "${STATUS} - Load: ${VALUE}%|'UPS Load'=${VALUE}%;$WARNVAL;$CRITVAL;;"
+	echo "${STATUS} - Load: ${VALUE}%|'UPS Load'=${VALUE}%;$WARNVAL;$CRITVAL;0;100"
 	;;
 timeleft)
 	if [ "$CRITVAL" = "" ]; then
@@ -301,7 +303,7 @@
 		STATUS="OK"
 		EXITVALUE=0
 	fi
-	echo "${STATUS} - Time Left: $VALUE Minutes|'minutes left'=${VALUE};$WARNVAL:;$CRITVAL:;;"
+	echo "${STATUS} - Time Left: $VALUE Minutes|'minutes left'=${VALUE};$WARNVAL;$CRITVAL;0;100"
 	;;
 esac
 
